Lib/test/test_metaclass.py

```diff
@@ -297,4 +297,5 @@ def load_tests(loader, tests, pattern):
 
 
 if __name__ == "__main__":
+    __name__ = "test.test_metaclass"
```

Comment:
Is there a way to configure doctests with a fake name instead of doing this? I don't know if it's the best solution for this. Otherwise, let's do something like:

```py
try:
    __name__ = "test.test_metaclass"  # to match doctest expectations
   unittest.main()
finally:
    __name__ = "__main__"
```

Reply:
what's the purpose of restoring `__name__` to it's old value? we are already at the end of the file anyway.

Reply:
Oups you're right :)